webapp/views/results-modal/script.js

```diff
@@ -1,16 +1,156 @@
-/* global $ */
+/* global $, get, deleteMethod */
 
-// eslint-disable-next-line no-unused-vars
-function display() {
-  const value = $('#results-select')[0].value;
-  if (value != '') {
-    $('#results-iframe').attr('src', '/output/' + value);
-  } else {
-    $('#results-iframe').attr('src', '');
+/**
+ * result modal class
+ */
+class ResultModal {
+  constructor(data = [], resultToDelete = '', currentView = 'ALL') {
+    this._data = data;
+    this.resultToDelete = resultToDelete;
+    this.currentView = currentView;
+  }
+
+  get dataAll() {
+    return this._data;
+  }
+  set dataAll(val) {
+    this._data = val;
+  }
+  get dataTimeMap() {
+    return this._data.filter(str => str.match(/^time/));
+  }
+  get dataQuery() {
+    return this._data.filter(str => str.match(/^query/));
+  }
+
+  remove(file) {
+    this.dataAll.splice(this.dataAll.indexOf(file), 1);
+  }
+
+  onClickTimeMap() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').removeClass('active');
+    $('#query-btn').removeClass('active');
+    $('#time-map-btn').addClass('active');
+
+    this.currentView = 'TIMEMAP';
+    this.dataTimeMap.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
+
+  onClickQuery() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').removeClass('active');
+    $('#query-btn').addClass('active');
+    $('#time-map-btn').removeClass('active');
+
+    this.currentView = 'QUERY';
+    this.dataQuery.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
+
+  onClickAll() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').addClass('active');
+    $('#query-btn').removeClass('active');
+    $('#time-map-btn').removeClass('active');
+
+    this.currentView = 'ALL';
+    this.dataAll.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
+
+  onClickDeleteAll() {
+    $('#delete-all-modal').show();
+  }
+
+  onClickDelete(btn) {
+    $('#delete-modal').show();
+
+    this.resultToDelete = btn.value;
+    $('#result-to-delete-span').text(resultModal.resultToDelete);
+  }
+
+  updateResults() {
+    get('/output', {}, res => new Promise(resolve => {
+      res.list.sort().reverse();
+      this.dataAll = res.list;
+
+      switch (this.currentView) {
+        case 'QUERY':
+          this.onClickQuery();
+          break;
+        case 'TIMEMAP':
+          this.onClickTimeMap();
+          break;
+        default:
+          this.onClickAll();
+          break;
+      }
+      resolve();
```

Comment:
Use an object to define the different behaviors:
```
{ timemap: this, query: that }
```
and use it to build the modal. It is the first and last time you need to write down anything specific about the different modules.

Reply:
Not sure what you mean. Could you explain a bit more?

Reply:
You've hardcoded the names of the modules, named functions after them and use a switch statement that relies on those names. The effect is that whenever the modules change or a new module is added, you need to adapt a lot of code here:
```
  get results() {
    return this._results;
  }
  get timeMapResults() {
    return this._results.filter(str => str.file.match(/^time/));
  }
  get queryResults() {
    return this._results.filter(str => str.file.match(/^query/));
  }
```
```
  onClickTimeMap() { ... }

  onClickQuery() { ... }

  onClickAll() ... }
```
```
      switch (this.currentView) {
        case 'QUERY':
          this.onClickQuery();
          break;
        case 'TIMEMAP':
          this.onClickTimeMap();
          break;
        default:
          this.onClickAll();
          break;
      }
```
Ideally you would have a single results getter and a single "onClick" function that takes a parameter indicating which module is active. The only things you need to know about each module are the id if the button and the regex used for filtering filenames . So:
```
modules = {
  timemap: { btnSelector: '#time-map-btn', regex: /^time/ },
  query: { btnSelector: '#query-btn', regex: /^query/ }
}
```
Now the "onClick" function becomes something like:
```
  filterByModule(module) {
    // set all buttons inactive
    // set $(module.btnSelector) active

    // this.currentView = module
    // append this.getResults(module).sort()
  }
```
and the switch becomes:
```
  filterByModule(this.currentView)
```
Isn't it much cleaner?

Reply:
aha! Got it!